app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/actions/ValidateAcceptHeaderAction.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.excisemovementcontrolsystemapi.controllers.actions
+
+import play.api.http.HeaderNames
+import play.api.libs.json.Json
+import play.api.mvc.Results.NotAcceptable
+import play.api.mvc.{ActionFilter, Request, Result}
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.ErrorResponse
+import uk.gov.hmrc.excisemovementcontrolsystemapi.utils.DateTimeService
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ValidateAcceptHeaderAction @Inject()(datetimeService: DateTimeService)(implicit ec: ExecutionContext) extends ActionFilter[Request] {
+
+  override val executionContext: ExecutionContext = ec
+
+  override def filter[A](request: Request[A]): Future[Option[Result]] = Future.successful {
+
+    val pattern = """^application/vnd[.]{1}hmrc[.]{1}(1|2){1}[.]0[+]{1}xml$""".r
```

Comment:
in here we validate the header. Here we only accepting application/vnd.hmrc.1.0+xml and application/vnd.hmrc.2.0+xml. Platform support API says that HMRC only support these two header and any other behaviour is unsupported and subject to change. However it will default to 1.0 if no Accept header is available and it will not be rejected. The vnd header is for routing to the correct version of the API. SO are we happy to validate this way or shell we also consider valid the following: application/xml? application/xml in here will be rejected.